WordPress/gutenberg
# Borders: Use new border control components in block support

Related:
- https://github.com/WordPress/gutenberg/issues/35602

Depends on: 
- https://github.com/WordPress/gutenberg/pull/39753
- https://github.com/WordPress/gutenberg/pull/37769
- https://github.com/WordPress/gutenberg/pull/38876


## Description

Updates the block support and global styles border panel to leverage the new [`BorderBoxControl`](https://github.com/WordPress/gutenberg/pull/38876) component for specifying individual borders per side. 

## How has this been tested?

1. Running unit tests
    - `npm run test-unit-php /var/www/html/wp-content/plugins/gutenberg/phpunit/block-supports/border-test.php`
    - `npm run test-unit-php /var/www/html/wp-content/plugins/gutenberg/phpunit/class-wp-theme-json-test.php` 
2. Manual testing via block editor
    1. Create a post, add a group block, and select it. 
    2. Trial different border configurations via the new `BorderBoxControl` in the inspector controls sidebar
    3. Save the post and confirm correct display on the frontend
    4. Switch back to the editor and add a dynamic block that has border support and repeat testing borders
3. Tested theme.json styling of individual borders
    - Update theme.json to configure various borders
```
        "core/group": {
		"border": {
			"left": {
				"color": "var(--wp--preset--color--primary)",
				"style": "dashed",
				"width": "3px"
			}
		}
	}
```
4. Manual testing of global styles
    1. Within the site editor's global styles sidebar, navigate to Blocks > Group > Layout
    2. Adjust borders and confirm the various group blocks in the editor update visually
    3. Save global styles and view on the frontend
5. Test other blocks that already opt into border support e.g. Search and Table blocks.
6. Test that blocks that already had borders via block support continue to display correctly after applying this PR

## Screenshots 

https://user-images.githubusercontent.com/60436221/160806276-85b14331-0b0b-430b-8f72-d7e3a7757d74.mp4


## Types of changes
Enhancement.

## Checklist:
